src/main/java/org/jfrog/hudson/ArtifactoryBuilder.java

```diff
@@ -192,6 +193,8 @@ public FormValidation doTestConnection(
                     version = artifactoryManager.getVersion();
                 } catch (UnsupportedOperationException uoe) {
                     return FormValidation.warning(uoe.getMessage());
+                } catch (ClientProtocolException e) {
+                    return FormValidation.error(e.getCause().getMessage());
```

Comment:
Let's add a comment here, explaining why we bother catching this specific exception separately, and why we pull the cause message, instead of the exception message.